### PR TITLE
Changes for Management.Sql to allow customers to add TDE keys and set TDE protector for managed instances

### PR DESCRIPTION
**Pull request by nivimsft**
## Description
Changes for Management.Sql to allow customers to add TDE keys and set TDE protector for managed instances

[Rest api spec PR](https://github.com/Azure/azure-rest-api-specs/pull/3762)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 1]
```diff
@@ -7,12 +7,13 @@
 		<PackageId>Microsoft.Azure.Management.Sql</PackageId>
 		<Description>Azure SQL Management SDK library</Description>
 		<AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-		<Version>1.20.0-preview</Version>
+		<Version>1.21.0-preview</Version>
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: revert, latest published version is still 1.19.0. see https://www.nuget.org/packages/Microsoft.Azure.Management.Sql

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 2]
```diff
@@ -7,12 +7,13 @@
...
 		<PackageTags>Microsoft Azure SQL Management;SQL;SQL Management;</PackageTags>
 		<PackageReleaseNotes>
 			<![CDATA[
 New features:
 - Support database sensitivity labels.
+- Added aps used to manage tde keys for ManagedInstance and to set TDE encryption protector for managed instances.
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: `- Added support for Managed Instance keys and encryption protectors.`

File: src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -22,5 +22,5 @@
 [assembly: AssemblyTitle("Microsoft Azure SQL Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure SQL.")]
 [assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.20.0.0")]
+[assembly: AssemblyFileVersion("1.21.0.0")]
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Properties/AssemblyInfo.cs]: revert

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: src/SDKs/SqlManagement/Sql.Tests/ServerKeyScenarioTests.cs [hunk 4]
```diff
@@ -53,12 +55,13 @@ public void TestCreateUpdateDropServerKey()
                 var keyList = sqlClient.ServerKeys.ListByServer(resourceGroup.Name, server.Name);
                 Assert.Equal(2, keyList.Count());
 
-                // Delete key
-                sqlClient.ServerKeys.Delete(resourceGroup.Name, server.Name, serverKeyName);
+                //TODO: Temporarily disabling this since delete operation is affected by a production bug.
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/ServerKeyScenarioTests.cs]: What is the production bug? We should avoid publishing an SDK that has known issues.

Comment (nivimsft) [reply src/SDKs/SqlManagement/Sql.Tests/ServerKeyScenarioTests.cs]: Hi Jared, 

The bug is in management service where the post condition checks are failing even on operation success. This issue seems to affect multiple APIs in prod. Management service team is currently investigating the issue. 
It is important to note that the failure occurs in an existing functionality (sqlClient.ServerKeys.Delete() method which was released last year) and is not caused by this change.

Comment (jaredmoo) [reply src/SDKs/SqlManagement/Sql.Tests/ServerKeyScenarioTests.cs]: Not related to managed instance keys.

**Review by jaredmoo — APPROVED** [review]

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs [hunk 5]
```diff
@@ -56,6 +56,8 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("Sql", "Jobs", "2017-03-01-preview"),
                 new Tuple<string, string, string>("Sql", "LongTermRetentionBackups", "2017-03-01-preview"),
                 new Tuple<string, string, string>("Sql", "ManagedDatabases", "2017-03-01-preview"),
+                new Tuple<string, string, string>("Sql", "ManagedInstanceEncryptionProtectors", "2017-10-01-preview"),
+                new Tuple<string, string, string>("Sql", "ManagedInstanceKeys", "2017-10-01-preview"),
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: Please run `msbuild build.proj /t:Build /p:Scope=SDKs\SqlManagement` and check in any changes generated

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 6]
```diff
@@ -13,6 +13,7 @@
 			<![CDATA[
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Looks like there are some additive changes. Please bump the version to `1.20.1-preview`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a couple of comments.

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj (hunk 6 above)
Comment (jaredmoo) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: 1.20 is not published yet.

Comment (dsgouda) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: No-op in that case

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs (hunk 5 above)
Comment (dsgouda) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: If there's nothing generated, we are good.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**